ccdaservice/ccda_gateway.php

```diff
@@ -7,140 +7,155 @@
+$parameterArray['view'] = 0;
+$parameterArray['recipient'] = 'patient'; // emr_direct or hie
+$parameterArray['site'] = $_SESSION ['site_id']; // set to an onsite portal user
+
+
+$server_url = resolveHost();
+// CCM returns viewable CCD html file or
+// zip containing a CCDA.xml, CCDA.html and cda.xsl
+$ccdaxml = portalccdafetching($pid, $server_url, $parameterArray, $dowhat);
+
+if ($dowhat === 'dl') {
+    header("Cache-Control: public");
+    header("Content-Description: File Transfer");
+    header("Content-Disposition: attachment; filename=SummaryofCare.zip");
+    header("Content-Type: application/download");
+    header("Content-Transfer-Encoding: binary");
+    echo $ccdaxml;
+    exit;
 }
 
-print_r($h . $ccdaxml . $h);
-//service_shutdown(1); //In ssmanager  0= terminate and disable 1 = soft=terminate but still active w/no restart, > 1 just restart based on B.S timer
+echo($ccdaxml);
+
 exit;
 
-function portalccdafetching($pid, $server_url, $parameterArray)
+function portalccdafetching($pid, $server_url, $parameterArray = [], $action = 'view')
 {
+    $parameters = '';
     $site_id = $_SESSION['site_id'];
-    $parameters = http_build_query($parameterArray); // future use
+    $url = $server_url . "/interface/modules/zend_modules/public/encounterccdadispatch/index?site=" .
+        urlencode($site_id) . "&me=" . urlencode(session_id()) .
+        "&param=1&view=1&combination=" . urlencode($pid) . "&recipient=patient";
+    if ($action === 'dl') {
+        $parameters = http_build_query($parameterArray);
+        $url = $server_url . "/interface/modules/zend_modules/public/encounterccdadispatch/index?me=" . urlencode(session_id());
+    }
     try {
         $ch = curl_init();
-        $url = $server_url . "/interface/modules/zend_modules/public/encounterccdadispatch/index?site=" . urlencode($site_id) . "&me=" . urlencode(session_id()) . "&param=1&view=1&combination=" . urlencode($pid) . "&recipient=patient";
         curl_setopt($ch, CURLOPT_URL, $url);
+        curl_setopt($ch, CURLOPT_POST, 1);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $parameters);
         curl_setopt($ch, CURLOPT_HEADER, 0); // set true for look see
         curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
         curl_setopt($ch, CURLOPT_COOKIESESSION, true);
         curl_setopt($ch, CURLOPT_COOKIEJAR, "cookie");
         curl_setopt($ch, CURLOPT_COOKIEFILE, "cookie");
-        //curl_setopt ($ch, CURLOPT_COOKIE, 'XDEBUG_SESSION=1'); // break on first line in public/index.php - uncomment and start any xdebug session and fetch a ccda in app.
-        curl_setopt($ch, CURLOPT_USERAGENT, "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)");
         curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-
         curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
         curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+        curl_setopt($ch, CURLOPT_COOKIE, 'XDEBUG_SESSION=1'); // debug break on first line in public/index.php
```

Comment:
is this a debugging thing?(ie. meant for production)

Reply:
yes, forgot